sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/optimizer/Optimizer.scala

```diff
@@ -152,6 +153,19 @@ abstract class Optimizer(sessionCatalog: SessionCatalog, conf: SQLConf)
 }
 
 /**
+ * Remove useless DISTINCT for MAX and MIN.
+ * This rule should be applied before RewriteDistinctAggregates.
+ */
+object EliminateDistinct extends Rule[LogicalPlan] {
+  override def apply(plan: LogicalPlan): LogicalPlan = plan transformExpressions  {
+    case AggregateExpression(max: Max, mode: AggregateMode, true, _) =>
```

Comment:
it's unclear what the "true" is. I'd either use named argument, or rewrite it to something like

```
case ae: AggregateExpression if ae.isDistinct =>
  ae.aggregateFunction match {
    case _: Max | _: Min => ae.copy(isDistinct = false)
  }
}
```